cpp/src/rolling/rolling.cu

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utilities/nvtx/nvtx_utils.h"
+#include "utilities/type_dispatcher.hpp"
+#include "rmm/thrust_rmm_allocator.h"
+#include <bitmask/legacy_bitmask.hpp>
+#include <cudf.h>
+#include <cub/cub.cuh>
+#include <memory>
+#include <stdio.h>
+#include <algorithm>
+
+#include <rolling.hpp>
+
+// allocate column
+#include <io/utilities/wrapper_utils.hpp>
+
+// basic aggregation classes from groupby
+#include <groupby/aggregation_operations.hpp>
```

Comment:
In order to unify the functors we use for the various aggregators, I'm going to remove `groupby/aggregation_operations.hpp` in favor of using https://github.com/rapidsai/cudf/blob/branch-0.7/cpp/src/utilities/device_atomics.cuh#L43

I plan to do this in https://github.com/rapidsai/cudf/pull/1478

Reply:
OK, given that we want to have rolling windows submitted ASAP I would just add a TODO here and we'll switch to the new API later.